src/term.rs

```diff
@@ -1298,6 +1298,28 @@ impl RichTerm {
             TraverseOrder::BottomUp => f(result, state),
         }
     }
+
+    /// Pretty print a term capped to a given max length (in characters). Useful to limit the size
+    /// of terms reported e.g. in typechecking errors. If the output of pretty printing is greater
+    /// than the bound, the string is truncated to `max_width` and the last character after
+    /// truncate is replaced by the ellipsis unicode character U+2026.
+    pub fn pretty_print_cap(&self, max_width: usize) -> String {
+        let output = format!("{}", self);
+
+        if output.len() <= max_width {
+            output
+        } else {
+            let (end, _) = output.char_indices().nth(max_width).unwrap();
+            let mut truncated = String::from(&output[..end]);
```

Comment:
Shouldn't we care about unicode here? What if the term is `Term::String("y̆es")` ?
`"y̆es"` is 3 graphemes but 4 chars...

Reply:
I tend to think that we don't really care about cutting off less characters that expected: it's mostly an helper to avoid going over `max_width`. So as long as the string is strictly smaller, I guess that's fine? To my knowledge, iterating over graphemes also requires an external crate (and blow the size of the executable because it needs to include Unicode tables), which doesn't seem to be worth the trouble.

Reply:
Fine by me